vnext/src/Libraries/Components/Button.windows.js

```diff
@@ -335,7 +335,62 @@ class Button extends React.Component<
     //  Platform.OS === 'android' ? TouchableNativeFeedback : TouchableOpacity;
     // Windows]
     // [Windows
-    if (Platform.OS === 'windows') {
+    if (Platform.OS === 'windows' && color) {
+      return (
+        <Touchable
+          accessibilityLabel={accessibilityLabel}
+          accessibilityRole="button"
+          accessibilityState={accessibilityState}
+          hasTVPreferredFocus={hasTVPreferredFocus}
+          nextFocusDown={nextFocusDown}
+          nextFocusForward={nextFocusForward}
+          nextFocusLeft={nextFocusLeft}
+          nextFocusRight={nextFocusRight}
+          nextFocusUp={nextFocusUp}
+          testID={testID}
+          disabled={disabled}
+          onPress={onPress}
+          tabIndex={tabIndex}
+          touchSoundDisabled={touchSoundDisabled}
+          underlayColor={PlatformColor('SolidBackgroundFillColorBaseBrush')}
+          onShowUnderlay={() => {
+            this.setState({pressed: true});
+          }}
+          onHideUnderlay={() => {
+            this.setState({pressed: false});
+          }}
+          onMouseEnter={() => {
+            if (!disabled) this.setState({hover: true});
+          }}
+          onMouseLeave={() => {
+            if (!disabled) this.setState({hover: false});
+          }}
```

Comment:
This is duplicating a lot of the logic for button without color.

What logic is different between the two? Are the differences confined to the style?

Reply:
Button logic between the two is the same, but styling logic between the two is quite different. I just pushed some changes which merges the logic for the button under one component, but it does introduce more conditional styling logic than when the cases are kept separate.